spec/controllers/my_facilities/drug_stocks_controller_spec.rb

```diff
@@ -0,0 +1,205 @@
+require "rails_helper"
+
+RSpec.describe MyFacilities::DrugStocksController, type: :controller do
+  let!(:facility_group_with_stock_tracked) { create(:facility_group) }
+  let!(:facilities_with_stock_tracked) { create_list(:facility, 3, facility_group: facility_group_with_stock_tracked) }
+  let!(:allowed_facility_for_manager) { facilities_with_stock_tracked.first }
+  let!(:disallowed_facility_for_manager) { facilities_with_stock_tracked.second }
+
+  let!(:protocol_drug) { create(:protocol_drug, stock_tracked: true, protocol: facility_group_with_stock_tracked.protocol) }
+  let!(:protocol_drug_2) { create(:protocol_drug, stock_tracked: true, protocol: facility_group_with_stock_tracked.protocol) }
+
+  let!(:facility_group) { create(:facility_group) }
+  let!(:facilities) { create_list(:facility, 3, facility_group: facility_group) }
+
+  let!(:power_user) { create(:admin, :power_user) }
+  let!(:manager) { create(:admin, :manager, :with_access, resource: allowed_facility_for_manager) }
+  let!(:report_viewer) { create(:admin, :viewer_reports_only, :with_access, resource: facility_group_with_stock_tracked) }
+
+  render_views
+
+  before do
+    Flipper.enable(:drug_stocks)
+  end
+
+  after do
+    Flipper.disable(:drug_stocks)
+  end
+
+  describe "GET #index" do
+    context "as power_user" do
+      it "returns a success response" do
+        sign_in(power_user.email_authentication)
+
+        get :index, params: {}
+        expect(response).to be_successful
+      end
+
+      it "only include facilities with tracked protocol drugs" do
```

Comment:
flaky

Reply:
Strange, I don't see anything in the test that makes this flaky.